ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/entity/PropertiesList.java

```diff
@@ -159,6 +254,112 @@
       FAIL_ON_DESERIALIZATION_ERROR_CONFIG
   );
 
+  public static final List<String> EditablePropertyList = ImmutableList.of(
```

Comment:
why are these all editable? only a handful of these should be editable right? can you add a link to the document that classifies the configs?

Reply:
I think @ableegoldman was saying that all configs in `StreamsConfig`, `ConsumerConfig`, `ProducerConfig`, `KsqlConfig` and `AdminClient` that aren't in the `internal` list above should be editable, which are all the configs i listed here. 

@ableegoldman do you have more insight here? Should any of these configs not be editable?

Reply:
See my response on slack, in sum though we should remove all of the ksql (and admin) configs from the allowlist.

Side note: the  `SINK_WINDOW_CHANGE_LOG_ADDITIONAL_RETENTION_MS_PROPERTY,` seems to actually be useless, it's never, I don't understand why there's a separate ksql config for this since it just corresponds to a StreamsConfig, in fact it's never handled by ksql so given the extra `SINK` prefix in the name relative to the  actual Streams config, I believe it wouldn't even work anyways. I guess I'll just file a ticket to clean that up on the side

Reply:
I tried to write a method to get all the config names from the classes directly instead of specifying them one by one (see line 362) but this list weirdly includes properties such as `sasl.mechanism` or `ssl.protocol` that are turned by `ConsumerConfig.configNames()`. I am assuming that we don't want them to be editable? Is there a way to exclude them perhaps or should we stick to the manual definition?